code/createDeviceSettingsTable.m

```diff
@@ -0,0 +1,757 @@
+function [TD_SettingsOut, Power_SettingsOut, FFT_SettingsOut, metaData] = createDeviceSettingsTable(folderPath)
```

Comment:
This file has a lot of redundant lines or lines that change very little from different sections of the code. While all the code contained in this function works, it should be cleaned up and broken up into local functions. 

One suggestion that I didn't know how to put down below. When building the out tables, it might be worth it to check if you have the same number of chunks in TD, power, and FFT before construction. Would be a good warning to throw, and would let the user know there is something weird about the data file.

Reply:
I agree, it's a bit tricky bcs an overloaded function could also get confusing with variable names being passed to generic placeholders etc. Not clear what you mean by "chunks?" you mean packets? There is no guarantee you will have the same number as FFT packets depend on the FFT interval.

Reply:
Sorry, chunks was a poor choice. I mean the same number of recording sessions.

Reply:
I agree that it will be tricky to separate out all the redundant code. Put that suggestion on the back burner.

Reply:
Agree that there is quite a bit of redundant code. Do you think it is worth creating a separate function for each TD, power, and FFT to create the `toAdd` struct? (Which then gets appended to the *_SettingsTable). I didn't do this initially because we'd be passing so much info into the function, but might make it more readable?

Reply:
Worth it in terms of readability? I'm hesitant to say it would be beneficial and I'm leaning towards inconsequential. I can only see the removal of incremental counter variable statements being the biggest thing. Like you said, a lot of data will be passed in the simplest implementation of a data entry function, meaning a lot of the statements will remain. 

Where I see most of the benefit of splitting up some of these routines and assignments is for code management and debugging. It would be easier to find the code that does a specific type of routine, and allow easier code modifications and testing.

Reply:
I went ahead and implemented the functions (one each for creating new rows for TD, power, and FFT) and another for adding that new entry to the table. See commits 737a3c2c3b37f6eaa4b6ed69309ffaa1d24e592b, 3c8ea9090e1316455c4edc6a477f0a7466c5bf42, and 70da1b46c39a6b76aa918dbbff30056472de76d3